src/app/routes/api/v3/forms/__tests__/public-forms.verification.routes.spec.ts

```diff
@@ -231,4 +239,325 @@ describe('public-forms.verification.routes', () => {
       expect(response.body).toEqual(expectedResponse)
     })
   })
+
+  describe('POST /forms/:formId/fieldverifications/:transactionId/fields/:fieldId/otp/generate', () => {
+    beforeEach(() => {
+      MockTwilio.messages.create.mockResolvedValue({
+        sid: 'mockSid',
+      })
+    })
+
+    it('should return 201 when parameters for email field are valid', async () => {
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockVerifiableFormId}/fieldverifications/${mockTransactionId}/fields/${mockEmailFieldId}/otp/generate`,
+        )
+        .send({
+          fieldType: VerifiableFieldType.Email,
+          answer: 'open@gov.sg',
+        })
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toBe(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 201 when parameters for mobile field are valid', async () => {
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockVerifiableFormId}/fieldverifications/${mockTransactionId}/fields/${mockMobileFieldId}/otp/generate`,
+        )
+        .send({
+          fieldType: VerifiableFieldType.Mobile,
+          // NOTE: a valid mobile number is +65 followed by ANY 8 digits
+          answer: '+6512345678',
+        })
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toBe(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 400 when fieldType is email but the provided email is not valid', async () => {
```

Comment:
again, is this correct given our discussion on the reliability of Joi validation? we don't want a situation where people are preventing from submitting because Joi hasn't updated their validation types